kernel/src/hil/digest.rs

```diff
@@ -156,11 +250,12 @@ pub trait DigestData<'a, const L: usize> {
 /// Computes a digest (cryptographic hash) over data provided through a
 /// separate trait.
 ///
-/// 'L' is the length of the 'u8' array to store the digest output.
-pub trait DigestHash<'a, const L: usize> {
+/// `DigestAlgorithm` is the type of digest computed and the data structure to
+/// store the computed digest.
+pub trait DigestHash<'a, D: DigestAlgorithm> {
     /// Set the client instance which will receive the `hash_done()`
     /// callback.
-    fn set_hash_client(&'a self, client: &'a dyn ClientHash<L>);
+    fn set_hash_client(&'a self, client: &'a dyn ClientHash<D>);
```

Comment:
Are you positive these `&'a self` are necessary? I find that's often an anti-pattern.